### ISSUE TITLE
Connection timeout

### ISSUE THREAD

**Issue by Fabian-Ser** [codetheweb/tuyapi#301]
**Describe the bug**
After running the cli command and after looking the https://github.com/codetheweb/tuyapi/issues/152 i get the output: 
× Device(s) failed to be registered!
Error: Timed out waiting for devices to connect.
    at TuyaLinkWizard.linkDevice (C:\Users\Stef Wolters\.npm-global\node_modules\@tuyapi\cli\node_modules\@tuyapi\link\index.js:117:17)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async link (C:\Users\Stef Wolters\.npm-global\node_modules\@tuyapi\cli\lib\link.js:47:19)

**To Reproduce**
1. Run the command: tuya-cli link --api-key <your api key> --api-secret <your api secret> --schema <your-schema/channel ID> --ssid <your WiFi name> --password <your WiFi password> --region us
2. Wait

**Expected behavior**
I Would connect my devices, but its not working

**Debug Output**
Post the output of your program/app/script when run with the `DEBUG` environment variable set to `*`.  Example: `DEBUG=* node test.js`.  Copy the output and paste it below (in between the code fences):

Ran: setx DEBUG *, same output as above 

**Desktop (please complete the following information):**
 - OS: [Windows 10]
 - OS Version [Home]
 - Node Version [12.16.1]

**Comment by codetheweb** [comment]
Do you have a link to the product page for the device that you're trying to connect?

**Comment by Fabian-Ser** [comment]
https://www.action.com/nl-nl/p/lsc-smart-connect-slimme-multicolor-ledlamp-3/

**Comment by scoobaspeaz** [comment]
I too am having this same exact issue. I would open up a different bug but I didn't want to duplicate.

Here is my device im trying to connect: https://www.walmart.com/ip/Monster-Smart-Google-Assistant-Multicolor-LED-Light-Strip-6-5-ft/541990457

**Comment by Fabian-Ser** [comment]
Any idea's?

**Comment by codetheweb** [comment]
Are either of you able to tell using Wireshark / router logs / physical indicators on the device whether or not the device connects to the WiFi network but fails to be fully configured?

**Comment by Fabian-Ser** [comment]
I'm not an expert at Wireshark, but I got more logs while running the command: 

https://gyazo.com/d22829e9ec8b84038644fc71bd5919cb

**Comment by Fabian-Ser** [comment]
- Registering devices(s)...  @tuyapi/link:wizard 0 devices returned by API. +102ms                                                 \ Registering devices(s)...  @tuyapi/link:wizard 0 devices returned by API. +77ms                                                    @tuyapi/link:wizard 0 devices returned by API. +116ms                                                                            | Registering devices(s)...  @tuyapi/link:wizard 0 devices returned by API. +121ms                                                 / Registering devices(s)...  @tuyapi/link:wizard 0 devices returned by API. +79ms                                                  - Registering devices(s)...  @tuyapi/link:wizard 0 devices returned by API. +88ms                                                    @tuyapi/link:wizard 0 devices returned by API. +84ms                                                                             \ Registering devices(s)...  @tuyapi/link:wizard 0 devices returned by API. +120ms                                                 | Registering devices(s)...  @tuyapi/link:wizard 0 devices returned by API. +107ms                                                 / Registering devices(s)...  @tuyapi/link:wizard 0 devices returned by API. +82ms                                                    @tuyapi/link:wizard 0 devices returned by API. +78ms                                                                             - Registering devices(s)...  @tuyapi/link:wizard 0 devices returned by API. +83ms                                                  \ Registering devices(s)...  @tuyapi/link:wizard 0 devices returned by API. +82ms                                                    @tuyapi/link:wizard 0 devices returned by API. +107ms                                                                            | Registering devices(s)...  @tuyapi/link:wizard 0 devices returned by API. +112ms                                                 / Registering devices(s)...  @tuyapi/link:wizard 0 devices returned by API. +108ms                                                 - Registering devices(s)...  @tuyapi/link:wizard 0 devices returned by API. +80ms                                                    @tuyapi/link:wizard 0 devices returned by API. +79ms                                                                             \ Registering devices(s)...  @tuyapi/link:wizard 0 devices returned by API. +82ms                                                  | Registering devices(s)...  @tuyapi/link:wizard 0 devices returned by API. +87ms                                                    @tuyapi/link:wizard 0 devices returned by API. +81ms                                                                             / Registering devices(s)...  @tuyapi/link:wizard 0 devices returned by API. +116ms                                                 - Registering devices(s)...  @tuyapi/link:wizard 0 devices returned by API. +108ms                                                   @tuyapi/link:wizard 0 devices returned by API. +80ms                                                                             \ Registering devices(s)...  @tuyapi/link:wizard 0 devices returned by API. +84ms                                                  | Registering devices(s)...  @tuyapi/link:wizard 0 devices returned by API. +113ms                                                 / Registering devices(s)...  @tuyapi/link:wizard 0 devices returned by API. +82ms                                                    @tuyapi/link:wizard 0 devices returned by API. +85ms                                                                             - Registering devices(s)...  @tuyapi/link:wizard 0 devices returned by API. +106ms                                                 \ Registering devices(s)...  @tuyapi/link:wizard 0 devices returned by API. +77ms                                                  | Registering devices(s)...  @tuyapi/link:wizard 0 devices returned by API. +114ms                                                   @tuyapi/link:wizard 0 devices returned by API. +115ms                                                                            / Registering devices(s)...  @tuyapi/link:wizard 0 devices returned by API. +88ms                                                  - Registering devices(s)...  @tuyapi/link:wizard 0 devices returned by API. +103ms                                                   @tuyapi/link:wizard 0 devices returned by API. +76ms                                                                             \ Registering devices(s)...  @tuyapi/link:wizard 0 devices returned by API. +76ms                                                  | Registering devices(s)...  @tuyapi/link:wizard 0 devices returned by API. +84ms                                                    @tuyapi/link:wizard 0 devices returned by API. +104ms                                                                            / Registering devices(s)...  @tuyapi/link:wizard 0 devices returned by API. +95ms                                                  - Registering devices(s)...  @tuyapi/link:wizard 0 devices returned by API. +79ms                                                    @tuyapi/link:wizard 0 devices returned by API. +81ms                                                                             / Registering devices(s)...  @tuyapi/link:wizard 0 devices returned by API. +37s

**Comment by codetheweb** [comment]
Sorry, there doesn't seem to be an image at that link.  If you took a screenshot of Wireshark, you instead can just export the capture as a .pcap and upload that.

**Comment by Fabian-Ser** [comment]
https://gyazo.com/f8c064004e753730cc2636bd1fc854aa

**Comment by Fabian-Ser** [comment]
Okay, seems like working now 🤔

**Comment by codetheweb** [comment]
👍

**Comment by scoobaspeaz** [comment]
Doesnt work for me yet. Am I doing something wrong lol

Here is what I do..I fire up tuya-cli  and it connect and waits for something to connect to. I then put my device in paring mode and wait.  The device blinks and blinks and blinks but eventually doesnt do anything. The tuya-cli  doesnt ever find anything and closes out with the message above.

**Comment by scoobaspeaz** [comment]
Does tuya-cli  need to be ran on a device that has a wifi card built into it? That part was never clear in the instructions to me at least. I am running it on an intel nuc which does have a wireless card but its not configured. The NUC is cabled to my router which the router has a functioning 2.4ghz band running.

**Comment by codetheweb** [comment]
You should put your device in pairing mode **before** you start the `tuya-cli` process.

In theory, it should work if your computer has a wired connection as long as your router is configured to broadcast packets.  In practice, it might be useful to try on a device with a WiFi adapter just to remove an additional potential point of failure.

**Comment by Fabian-Ser** [comment]
I used the following steps to get to the end result:


1. Use another device (with wireless connection), the device I had used previously only worked with an internet cable, so my PC did not connect to my lamps.

2. Put your tuya device into pairing mode

3. Run the command again

4. The lights will stop blinking and will now display a result in the console with all the necessary information

Hope this Will help u!
